YouTube/Classes/ViewControllers/HistoryViewController/HistoryViewController.swift

```diff
@@ -7,21 +7,42 @@
 //
 
 import UIKit
+import RealmSwift
 
 class HistoryViewController: BaseViewController {
 
     @IBOutlet weak private var historyTableView: UITableView!
+    private var videos: Results<History>!
+    private var date: [String] = []
+    private var listVideo: [[History]] = []
+
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        addNotification()
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
     }
+    func addNotification() {
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(addNewVideo), name: AppDefine.AddVideoToHistory, object: nil)
+    }
+    override func viewWillAppear(animated: Bool) {
+        date = []
+        loadData()
```

Comment:
why do you add it in here ? You call it 2 times in viewDidLoad and viewWillAppear